pkg/kubelet/kubelet_pods.go

```diff
@@ -135,7 +136,37 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			return nil, err
 		}
 		if mount.SubPath != "" {
+			fileinfo, err := os.Lstat(hostPath)
+			if err != nil {
+				return nil, err
+			}
+			gid := fileinfo.Sys().(*syscall.Stat_t).Gid
+			perm := fileinfo.Mode()
+
 			hostPath = filepath.Join(hostPath, mount.SubPath)
+
+			// If fsGroup volume ownership was set for the mounter path, set it for the sub path too
+			fsGroup := pod.Spec.SecurityContext.FSGroup
+			if gid == uint32(*fsGroup) {
+				if err := os.MkdirAll(hostPath, perm); err != nil {
+					glog.Errorf("failed to mkdir:%s", hostPath)
+					return nil, err
+				}
+				volume.SetVolumeOwnership(vol.Mounter, fsGroup, hostPath)
+
+				// If umask is preventing us from making the sub path have the same permissions as the
+				// mounter path, chmod the sub path
+				fileinfo, err := os.Lstat(hostPath)
```

Comment:
SetVolumeOwnership ORs the passed in path's existing permissions with 0660. So say I begin with an emptyDir mount path /a with perm bits 0777. Then I Mkdir the sub path /a/b, b has 0750. Next I do SetVolumeOwnership on b and b will come out of it with only 0750|0660=0760. So to make b have the same 0777 perm bits as its parent mount path a (which is what the user expects), I need to explicitly chmod it.